server/schedule/region_scatterer.go

```diff
@@ -323,6 +323,14 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string) *
 		scatterWithSameEngine(peers, ctx)
 	}
 
+	// If the number of target peers is too small, change it back to the original.
```

Comment:
I think it's better add a filter that can filter other peer stores in `selectCandidates `, like:
```
func (r *RegionScatterer) selectCandidates(region *core.RegionInfo, sourceStoreID uint64, selectedStores map[uint64]struct{}, context engineContext) []uint64 {
...
excludedPeersStores := make(map[uint64]struct{})
	for _, peer := range region.GetPeers() {
		if peer.GetId() != sourceStoreID {
			excludedPeersStores[peer.GetStoreId()] = struct{}{}
		}
	}
	filters := []filter.Filter{
		filter.NewExcludedFilter(r.name, nil, selectedStores),
		filter.NewExcludedFilter(r.name, nil, excludedPeersStores),
	}
...
}

```
This will make the region distribution more dispersed than the current strategy.

Reply:
It is necessary to have some regions unchanged.

Reply:
the goal is scatter them.

Reply:
itself is excluded, it will choose the smallest of the remaining ones resulting in redundant scheduling.